### ISSUE TITLE
New test pedigree simulation class

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#1812]
We want to add a new test class (TestSimulateThroughPedigree I guess) which will run the simulations through pedigrees for a number of different examples. I'd imagine the way it'll work is to have something like

```python
class TestSimulateThroughPedigree:
    def verify(self, input_tables):
        # Run the simulation through a pedigree defined by the input_tables, and verify the output
   
    @pytest.mark.parametrize("num_founders", [2, 3, 10, 100])
    def test_shallow(self, num_founders):
          # run pedigree simulation, call verify

    @pytest.mark.parametrize("num_founders", [2, 3, 10, 20])
    def test_deep(self, num_founders):
          # run pedigree simulation, call verify
    
   @pytest.mark.parametrize("num_founders", [2, 3, 10, 20])
    def test_many_children(self, num_founders)
          # Tweak the num_children_prob parameter

    def test_something_else...
```

So, we want to gather together a bunch of different examples, and make sure that we're simulating through this pedigree correctly. If we can come up with hand-crafted weird examples, then all the better.

The verification code can probably be based directly on what we're doing here: https://github.com/tskit-dev/msprime/blob/b13e7d948c52f798ef04d84f865a8acd6d6c6b23/tests/test_algorithms.py#L194

How does this sounds @nikbaya?

**Comment by nikbaya** [comment]
@jeromekelleher Should I be creating a new test_simulate_through_pedigree.py module to put this class in? Or put it in test_pedigree.py?

**Comment by jeromekelleher** [comment]
This can go in ``test_pedigree.py``

**Comment by nikbaya** [comment]
@jeromekelleher When you say "hand-crafted", does that mean it's ok to create a pedigree outside the rigid confines of the `simulate_pedigree` function?

**Comment by jeromekelleher** [comment]
Sure - it's simpler to come up with small examples of weird pedigrees that to try and simulate them.

**Comment by jeromekelleher** [comment]
Closed in #1819
